source-oracle/capture_test.go

```diff
@@ -0,0 +1,369 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/bradleyjkemp/cupaloy"
+	"github.com/estuary/connectors/sqlcapture/tests"
+	"github.com/sirupsen/logrus"
+)
+
+func TestViewDiscovery(t *testing.T) {
+	var unique = "18110541"
+	var tb, ctx = oracleTestBackend(t), context.Background()
+	var tableName = tb.CreateTable(ctx, t, unique, "(id INTEGER PRIMARY KEY, grp INTEGER, data VARCHAR(2000))")
+
+	var view = fmt.Sprintf(`"t%s"`, unique+"_simpleview")
+	tb.Query(ctx, t, false, fmt.Sprintf(`DROP VIEW %s`, view))
+	tb.Query(ctx, t, true, fmt.Sprintf(`CREATE VIEW %s AS SELECT id, data FROM %s WHERE grp = 1`, view, tableName))
+	t.Cleanup(func() {
+		logrus.WithField("view", view).Debug("dropping view")
+		tb.Query(ctx, t, false, fmt.Sprintf(`DROP VIEW %s`, view))
+	})
+
+	var bindings = tb.CaptureSpec(ctx, t).Discover(ctx, t, regexp.MustCompile(regexp.QuoteMeta(strings.TrimPrefix(tableName, "test."))))
+	for _, binding := range bindings {
+		logrus.WithField("name", binding.RecommendedName).Debug("discovered stream")
+		if strings.Contains(string(binding.RecommendedName), "_simpleview") {
+			t.Errorf("view returned by catalog discovery")
+		}
+		if strings.Contains(string(binding.RecommendedName), "_matview") {
+			t.Errorf("materialized view returned by catalog discovery")
+		}
+	}
+}
+
+func TestAllTypes(t *testing.T) {
+	var unique = "18110541"
+	var tb, ctx = oracleTestBackend(t), context.Background()
+	var typesAndValues = [][]any{
+		[]any{"nvchar2", "NVARCHAR2(2000)", "nvarchar2 value with unicode characters ❤️ \\ 🔥️'')"},
```

Comment:
The `[]any` type specification for each item in the `[][]any` is redundant and can be removed - flagging this since `gopls` flags it.